### DISCUSSION TITLE
Implement some modal editing mode

### DISCUSSION THREAD

**Discussion by soomtong** [charmbracelet/bubbletea#390] [Q&A] [answered]
Hi, I think bubble with bubbletea can make a TUI app that I imagine. 
but am having trouble implementing modal mode with textarea or textinput component.

I hope someone guides me.

I was thinking of simply switch a pane into a textinput (or area) component and views component.
Already the views component provides a j/k key moving like Vim.

I'd like to ask if this is the proper way?

Thanks in advance.

**Comment by maaslalani** [comment, marked as answer]
Hey @soomtong, this is a great idea! I believe the way to implement this is through our key component bubble https://github.com/charmbracelet/bubbles/blob/master/key/key.go

I believe you would define the key bindings and actions for normal mode and insert mode and then define a key (escape / i) to switch between the two key bindings.

**Reply by soomtong** [reply]
I think this will help me in learning go lang by TUI app lover. and I look forward to the day when introduce it happily.
Thanks a lot!
